62_CAD/fragment_shader.hlsl

```diff
@@ -372,6 +372,23 @@ float32_t4 calculateFinalColor<true>(const uint2 fragCoord, const float localAlp
     return col;
 }
 
+// TODO: place this MSDF code elsewhere
+float median(float r, float g, float b) {
+    return max(min(r, g), min(max(r, g), b));
+}
+
+float screenPxRange(float2 screenSpaceSizePixels, float2 distanceFieldSizePixels) {
+    float2 unitRange = MsdfPixelRange / distanceFieldSizePixels;
+    return max(0.5 * dot(unitRange, screenSpaceSizePixels), 1.0);
+}
+
+float msdfOpacity(float3 msd, float2 screenPxRangeValue) {
+    float sd = median(msd.r, msd.g, msd.b);
+    float screenPxDistance = screenPxRangeValue * (sd - 0.5);
+    float opacity = clamp(screenPxDistance + 0.5, 0.0, 1.0);
```

Comment:
Don't output opacity, output screen space distances and do AA with smooth step like we adjust do.
This is literally a sdf function, but reads from texture

Reply:
Done